libraries/AP_Camera/AP_Camera_Servo.cpp

```diff
@@ -61,19 +61,31 @@ bool AP_Camera_Servo::trigger_pic()
 
 bool AP_Camera_Servo::set_zoom(ZoomType zoom_type, float zoom_value)
 {
-    if (zoom_type == ZoomType::RATE) {
-        zoom_current_rate = zoom_value;
-        return true;
+    switch (zoom_type) {
+        case ZoomType::RATE:
+            zoom_current_rate = zoom_value;
+            return true;
+        case ZoomType::PCT:
+            // expects to receive a value between 0 and 100
+            SRV_Channels::set_output_scaled(SRV_Channel::k_cam_zoom, constrain_float(zoom_value * 10, 0, 1000));
```

Comment:
This is unclear - if it's expecting a 0-100 value, why are you providing a 0-1000 constraint?